core/src/docs/rfcs/2602_object_versioning.md

```diff
@@ -0,0 +1,102 @@
+Proposal Name: object versioning
+Start Date: 2023-07-06
+RFC PR: https://github.com/apache/incubator-opendal/pull/2602
+Tracking Issue: (leave this empty)
+
+# Summary
+
+This proposal describes the object versioning (or object version control) feature of OpenDAL.
+
+# Motivation
+
+Object versioning is a common feature in many storage services.
```

Comment:
Please note that the reader may not be familiar with storage services like you are. Therefore, the concept of `object version` may not be *common* to them.